### ISSUE TITLE
Never throw NullReferenceExeptions by yourself

### ISSUE THREAD

**Issue by nicolaiw** [Mr-Markus/ZigbeeNet#15]
https://github.com/Mr-Markus/ZigbeeNet/blob/f78f6e5ad04d1a072ef6570782ff76be5aa94004/ZigbeeNet/CC/CCZnp.cs#L407
NullReferenceExeptions should never be thrown manually.

**Comment by Mr-Markus** [comment]
Yes, I agree. Better would be Exeptions like `throw new ArgumentNullException(nameof(buffer));`

**Comment by nicolaiw** [comment]
In this case I would throw an ArgumentException because we also throw an exception when the length is equals 0.
